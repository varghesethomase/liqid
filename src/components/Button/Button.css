.button {
  background-color: #ffffff;
  border-radius: 4px;
  display: inline-block;
  box-shadow: 0 4px 6px rgba(50, 50, 93, 0.11), 0 1px 3px rgba(0, 0, 0, 0.08);
  font-size: 15px;
  height: 40px;
  line-height: 40px;
  padding: 0 14px;
  text-decoration: none;
  transition: all 0.15s ease;
  font-weight: 500;
  cursor: pointer;
  border: 0;
  outline: 0;
}

.button:hover {
  transform: translateY(-1px);
  box-shadow: 0 7px 14px rgba(50, 50, 93, 0.1), 0 3px 6px rgba(0, 0, 0, 0.08);
}
