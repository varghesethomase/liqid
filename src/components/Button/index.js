import React from 'react';
import PropTypes from 'prop-types';

import './Button.css';

const Button = props => (
  <button
    className={`button ${props.className}`}
    disabled={props.fieldValue === ''}
    onClick={props.buttonClick}
  >
    {props.children}
  </button>
);

Button.defaultProps = {
  className: ''
};

Button.propTypes = {
  buttonClick: PropTypes.func.isRequired,
  children: PropTypes.oneOfType([PropTypes.string, PropTypes.element])
    .isRequired,
  className: PropTypes.string.isRequired
};

export default Button;
