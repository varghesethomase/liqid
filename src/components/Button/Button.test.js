import React from 'react';
import { shallow } from 'enzyme';
import Button from './index';

it('renders the button', () => {
  const onButtonClick = jest.fn();
  const wrapper = shallow(<Button buttonClick={onButtonClick}>HELLO</Button>);
  expect(wrapper).toMatchSnapshot();
});
