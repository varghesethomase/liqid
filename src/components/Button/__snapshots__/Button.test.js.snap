// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the button 1`] = `
<button
  className="button "
  disabled={false}
  onClick={[Function]}
>
  HELLO
</button>
`;
